### DISCUSSION TITLE
Enhancement: Custom proxy

### DISCUSSION THREAD

**Discussion by qhs200312** [danny-avila/LibreChat#1804] [Q&A]
### What features would you like to see added?

Is it possible to add the function of customizing the proxy server address?

### More details

You can use a custom proxy server to access openai's services. Some areas do not support direct connection to openai.

### Which components are impacted by your request?

General, UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Yup you can customize it for the OpenAI endpoint: https://docs.librechat.ai/install/configuration/dotenv.html#openai

also you can configure OpenAI spec compatible endpoints here: https://docs.librechat.ai/install/configuration/custom_config.html

**Comment by qhs200312** [comment]
It would be even better if you could add a function to the front-end page of the website to allow users to enter the address of the proxy server themselves.

**Reply by danny-avila** [reply]
Yup it's already an open issue and will implement it soon: https://github.com/danny-avila/LibreChat/issues/1017
